components/org.wso2.carbon.identity.api.user.push.device.handler.v1/src/main/resources/push-auth.yaml

```diff
@@ -0,0 +1,780 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+      tags:
+        - admin
+
+    get:
+      description: |
+        This API is used by admin set to retrieve a specific device.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/manage/identity/user/push_divice_mgt/view <br/>
+          <b>OAuth2 Scopes:</b>
+         * internal_identity_mgt_view
+      summary: |
+        Returns Specific Device.
+      parameters:
+        - name: "user-id"
+          in: "path"
+          description: "ID of user"
+          required: true
+          schema:
+            type: "string"
+
+        - name: "deviceId"
+          in: "path"
+          description: "ID of device to return"
+          required: true
+          schema:
+            type: "string"
+      responses:
+        200:
+          description: Specific device selected by the user
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DeviceDTO'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+
+      tags:
+        - admin
+
+  /{user-id}/push-auth/devices:
+    get:
+      description: |
+        This API is used by admins to retrieve all devices registered under a user.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/manage/identity/user/push_divice_mgt/list <br/>
+          <b>OAuth2 Scopes:</b>
+         * internal_identity_mgt_view
+      summary: |
+        Returns Devices of a user.
+      parameters:
+        - name: "user-id"
+          in: "path"
+          description: "ID of user"
+          required: true
+          schema:
+            type: "string"
+      responses:
+        200:
+          description: All registered devices of the user
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AllDevices'
+        400:
```

Comment:
What could cause a 400 response for the API?

Reply:
Removed